src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = (string) $this->_conn->getDatabase();
+
+        $columnsData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseColumns($currentDatabase),
+            'TABLE_NAME'
+        );
+        $foreignKeysData = $this->groupObjectRecordsByTable(
+            $this->getDatabaseForeignKeys($currentDatabase),
+            'TABLE_NAME'
+        );
+        $indexesData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseIndexes($currentDatabase),
+            'TABLE_NAME'
+        );
+
+        $tables = [];
+
+        foreach (array_keys($columnsData) as $tableName) {
+            $columns = $this->_getPortableTableColumnList(
+                $tableName,
+                '',
+                $columnsData[$tableName]
+            );
+
+            $foreignKeys = [];
+            if (! empty($foreignKeysData[$tableName])) {
+                $foreignKeys = $this->_getPortableTableForeignKeysList(
+                    $foreignKeysData[$tableName]
+                );
+            }
+
+            $indexes = [];
+            if (! empty($indexesData[$tableName])) {
+                $indexes = $this->_getPortableTableIndexesList(
+                    $indexesData[$tableName],
+                    $tableName
+                );
+            }
+
+            $table = new Table($tableName, $columns, $indexes, [], $foreignKeys, []);
+            if (! empty($columnsData[$tableName][0]['TABLE_COMMENTS'])) {
+                $table->addOption('comment', $columnsData[$tableName][0]['TABLE_COMMENTS']);
+            }
+
+            $tables[] = $table;
+        }
+
+        return $tables;
+    }
+
+    /**
+     * Gets all column definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseColumns(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT C.OWNER,
+                 C.TABLE_NAME,
+                 C.COLUMN_NAME,
+                 C.DATA_TYPE,
+                 C.DATA_DEFAULT,
+                 C.DATA_PRECISION,
+                 C.DATA_SCALE,
+                 C.CHAR_LENGTH,
+                 C.DATA_LENGTH,
+                 C.NULLABLE,
+                 D.COMMENTS,
+                 E.COMMENTS TABLE_COMMENTS
+            FROM ALL_TAB_COLUMNS C
+       LEFT JOIN ALL_TAB_COMMENTS E ON E.OWNER = C.OWNER AND E.TABLE_NAME = C.TABLE_NAME
```

Comment:
It doesn't look like table comments belong here. It should be a separate query.

Reply:
done